src/vs/workbench/parts/codeEditor/electron-browser/simpleWidgetEditor.ts

```diff
@@ -0,0 +1,90 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IEditorOptions } from 'vs/editor/common/config/editorOptions';
+import { ICodeEditorWidgetOptions } from 'vs/editor/browser/widget/codeEditorWidget';
+
+// Allowed Editor Contributions:
+import { MenuPreventer } from 'vs/workbench/parts/codeEditor/electron-browser/menuPreventer';
+import { SelectionClipboard } from 'vs/workbench/parts/codeEditor/electron-browser/selectionClipboard';
+import { ContextMenuController } from 'vs/editor/contrib/contextmenu/contextmenu';
+import { SuggestController } from 'vs/editor/contrib/suggest/suggestController';
+import { SnippetController2 } from 'vs/editor/contrib/snippet/snippetController2';
+import { TabCompletionController } from 'vs/workbench/parts/snippets/electron-browser/tabCompletion';
+
+export class SimpleWidgetEditorConfig {
+
+	public static getCodeEditorWidgetOptions(): ICodeEditorWidgetOptions {
+		return {
+			isSimpleWidget: true,
+			contributions: [
+				MenuPreventer,
+				SelectionClipboard,
+				ContextMenuController,
+				SuggestController,
+				SnippetController2,
+				TabCompletionController,
+			]
+		};
+	}
+
+	public static getEditorOptions(style: 'editor' | 'htmlinput', ariaLabel?: string): IEditorOptions {
```

Comment:
I'd rather split this into 2 separate functions. How about `getEditorOptions` and `getEditorAsInputBoxOptions` ?